paris_rollins.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/python2.6
+
+# Run up to a configurable limit of simultaneous paris-traceroutes, back
+# towards recent M-Lab client IP addresses told to us via SideStream.
+
+# TODO(joshb): this script is written to minimize external dependencies.
+# Later versions of subprocess include built in timeout handling, but we
+# can't use them because the M-Lab platform doesn't have them. This should
+# be revisited if the M-Lab platform is upgraded.
+
+# TODO(joshb): this version just provides a function to run paris-traceroute.
+# Next step is to use multiprocessing to run multiple traceroutes and poll
+# the Web100 agent 
+
+import os
+import multiprocessing
+import subprocess
+import sys
+import time
+
+# What binary to use for paris-traceroute
+PARIS_TRACEROUTE = '/usr/local/bin/paris-traceroute'
+# paris-traceroute is run at this nice level, to minimize impact on the host.
+WORKER_NICE = 19
+# paris-traceroute should take no longer than this to complete (timed out,
+# partial results will be discarded).
+WORKER_TIMEOUT = 20 
+# Maximum number of paris-traceoutes to run simultaneously (requests to run
+# more will be discarded).
+MAX_WORKERS = 10
+
+
+def log_worker(message):
+  print time.strftime('%Y%m%d %T %%s', time.gmtime(time.time())) % message
+
+
+def make_log_file_name(log_time, log_file_root, remote_ip, remote_port,
+                       local_ip, local_port):
+  log_file_relative = (time.strftime('%Y/%m/%d/%Y%m%dT%TZ%%s.paris',
+                                     time.gmtime(log_time)) %
```

Comment:
how about
```(time.strftime('%Y/%m/%d/%Y%m%dT%TZ', time.gmtime(log_time)) + 
    ('-'.join((remote_ip, str(remote_port), local_ip, str(local_port)))  
    + '.paris')```
? That way you don't have to double-escape the `%s` and it's a little more clear what's going on.